lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -475,6 +479,51 @@ def __init__(
             charm.on[self.relation_name].relation_broken, self._on_relation_broken
         )
 
+    def _handle_changed_secrets(self, event) -> None:
+        """Update operator and backup user passwords when rotation occurs.
+
+        Changes in secrets do not re-trigger a relation changed event, so it is necessary to listen
+        to secret changes events.
+        """
+        if (
+            not self.charm.unit.is_leader()
+            or not event.secret.label
+            or not self.model.get_relation(self.relation_name)
+        ):
+            return
+
+        # only one related config-server is possible
+        config_server_relation = self.charm.model.relations[self.relation_name][0]
```

Comment:
nit: since we can only have one relation, perhaps: 
`config_server_relation = self.model.get_relation(self.relation_name)`?